src/lazy/encoder/binary/v1_1/value_writer.rs

```diff
@@ -155,29 +156,34 @@ impl<'value, 'top> BinaryValueWriter_1_1<'value, 'top> {
     // TODO: write_f16(...)
 
     pub fn write_f32(mut self, value: f32) -> IonResult<()> {
-        if value == 0f32 && !value.is_sign_negative() {
-            self.push_byte(0x5A);
-            return Ok(());
+        match value.smallest_repr() {
```

Comment:
When we add support for f16, we could also DRY it up like this:
```rust
fn write_float_repr(mut self, value: FloatRepr) -> IonResult<()> {
    match value {
        // ...
    }
    Ok(())
}

pub fn write_f16(mut self, value: f16) -> IonResult<()> {
    self.write_float_repr(value.smallest_repr())
}

pub fn write_f32(mut self, value: f32) -> IonResult<()> {
    self.write_float_repr(value.smallest_repr())
}

pub fn write_f64(mut self, value: f64) -> IonResult<()> {
    self.write_float_repr(value.smallest_repr())
}
```

Reply:
I was thinking the same thing 👍